custom_components/pyscript/state.py

```diff
@@ -113,6 +115,12 @@ def set(cls, var_name, value, new_attributes=None, **kwargs):
             new_attributes.update(kwargs)
         _LOGGER.debug("setting %s = %s, attr = %s", var_name, value, new_attributes)
         cls.notify_var_last[var_name] = str(value)
+
+        if var_name.startswith("pyscript."):
+            # have this var tracked for restore
+            restore_data = await RestoreStateData.async_get_instance(cls.hass)
+            restore_data.async_restore_entity_added(var_name)
+
```

Comment:
How expensive is it to call these functions every time a pyscript state variable is set?  Would it be better to only do it the first time each state variable is set?

Reply:
I'm not very experienced with the Home Assistant codebase or python async code.

The `async_restore_entity_added` simply puts the entry on a `set`, so it isn't expensive at all. The get_instance method I'm not sure, I expect it to be cheap due to the singleton? https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/restore_state.py#L65

Reply:
I've since introduced our own set to track persisted states, so this code block will only evaluate once.